spark-operator/src/main/java/org/apache/spark/k8s/operator/reconciler/SparkReconcilerUtils.java

```diff
@@ -0,0 +1,242 @@
+import static org.apache.spark.k8s.operator.utils.SparkExceptionUtils.isConflictForExistingResource;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.DeletionPropagation;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.javaoperatorsdk.operator.api.reconciler.DeleteControl;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.spark.k8s.operator.BaseResource;
+import org.apache.spark.k8s.operator.Constants;
+import org.apache.spark.k8s.operator.SparkApplication;
+import org.apache.spark.k8s.operator.utils.ModelUtils;
+import org.apache.spark.k8s.operator.utils.Utils;
+
+@Slf4j
+public class SparkReconcilerUtils {
+
+  public static Map<String, String> commonOperatorResourceLabels() {
+    Map<String, String> labels = new HashMap<>();
+    labels.put(LABEL_RESOURCE_NAME, OPERATOR_APP_NAME.getValue());
+    return labels;
+  }
+
+  public static Map<String, String> defaultOperatorConfigLabels() {
+    Map<String, String> labels = new HashMap<>(commonOperatorResourceLabels());
+    labels.put("app.kubernetes.io/component", "operator-dynamic-config-overrides");
+    return labels;
+  }
+
+  public static Map<String, String> commonManagedResourceLabels() {
+    Map<String, String> labels = new HashMap<>();
+    labels.put(LABEL_SPARK_OPERATOR_NAME, OPERATOR_APP_NAME.getValue());
+    return labels;
+  }
+
+  public static Map<String, String> sparkAppResourceLabels(final SparkApplication app) {
+    return sparkAppResourceLabels(app.getMetadata().getName());
+  }
+
+  public static Map<String, String> sparkAppResourceLabels(final String appName) {
+    Map<String, String> labels = commonManagedResourceLabels();
+    labels.put(Constants.LABEL_SPARK_APPLICATION_NAME, appName);
+    return labels;
+  }
+
+  public static Map<String, String> driverLabels(final SparkApplication sparkApplication) {
+    Map<String, String> labels = sparkAppResourceLabels(sparkApplication);
+    labels.put(Constants.LABEL_SPARK_ROLE_NAME, LABEL_SPARK_ROLE_DRIVER_VALUE);
+    return labels;
+  }
+
+  public static Map<String, String> executorLabels(final SparkApplication sparkApplication) {
+    Map<String, String> labels = sparkAppResourceLabels(sparkApplication);
+    labels.put(Constants.LABEL_SPARK_ROLE_NAME, LABEL_SPARK_ROLE_EXECUTOR_VALUE);
+    return labels;
+  }
+
+  public static Set<String> getWatchedNamespaces() {
```

Comment:
Please move this to `./spark-operator/src/main/java/org/apache/spark/k8s/operator/utils/Utils.java`.

Reply:
Created https://github.com/apache/spark-kubernetes-operator/pull/23 top include this refactoring along with the breakdown